modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -62,73 +60,85 @@ const updateAdvancedProps = (modifier: DomModifier, data: CellData) => {
   modifier.setFormat('tablecellborderwidth', Util.addPxSuffix(data.borderwidth));
 };
 
-// NOTES:
-
-// When applying to a single cell, values can be falsy. That is
-// because there should be a consistent value across the cell
-// selection, so it should also be possible to toggle things off.
+/*
+  NOTES:
 
-// When applying to multiple cells, values must be truthy to be set.
-// This is because multiple cells might have different values, and you
-// don't want a blank value to wipe out their original values. Note,
-// how as part of this, it doesn't remove any original alignment before
-// applying any specified alignment.
+  When applying to a single cell, values can be falsy. That is
+  because there should be a consistent value across the cell
+  selection, so it should also be possible to toggle things off.
 
-const applyCellData = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[], oldData: CellData, data: CellData) => {
+  When applying to multiple cells, values must be truthy to be set.
+  This is because multiple cells might have different values, and you
+  don't want a blank value to wipe out their original values. Note,
+  how as part of this, it doesn't remove any original alignment before
+  applying any specified alignment.
+ */
+const applyStyleData = (editor: Editor, cells: SelectedCell[], data: CellData) => {
   const isSingleCell = cells.length === 1;
+  Arr.each(cells, (item) => {
+    const cellElm = item.element;
+    const modifier = isSingleCell ? DomModifier.normal(editor, cellElm) : DomModifier.ifTruthy(editor, cellElm);
+    const colModifier = item.column.map((col) =>
+      isSingleCell ? DomModifier.normal(editor, col) : DomModifier.ifTruthy(editor, col)
+    ).getOr(modifier);
+
+    updateSimpleProps(modifier, colModifier, data);
+
+    if (hasAdvancedCellTab(editor)) {
+      updateAdvancedProps(modifier, data);
+    }
+
+    // Remove alignment
+    if (isSingleCell) {
+      Styles.unApplyAlign(editor, cellElm);
+      Styles.unApplyVAlign(editor, cellElm);
+    }
+
+    // Apply alignment
+    if (data.halign) {
+      Styles.applyAlign(editor, cellElm, data.halign);
+    }
+
+    // Apply vertical alignment
+    if (data.valign) {
+      Styles.applyVAlign(editor, cellElm, data.valign);
+    }
+  });
+};
 
+const applyStructureData = (editor: Editor, data: CellData) => {
+  // Switch cell type if applicable. This will also fire a table modified event for the structure change
```

Comment:
Can we slightly reword this comment to better explain we skip firing an event here before the `mceTableCellType` command will fire the relevant event for us?

Maybe

```suggestion
  // Switch cell type if applicable. We don't need to fire a `TableModified` event here as `mceTableCellType` will also fire a table modified event for the structure change
```

Reply:
Oops forgot to update this when I made the change to not fire the events. Thanks for the catch!

Reply:
Also worth noting the description in the comment and suggestion is wrong. It needs to say we tell the command to not fire the event because we batch it and fire it after all cell updates.

Reply:
Updated